DQM/L1TMonitor/src/L1TStage2EMTF.cc

```diff
@@ -189,118 +345,265 @@ void L1TStage2EMTF::analyze(const edm::Event& e, const edm::EventSetup& c) {
+      {{CO.ME3_1(), CO.ME3_2(), CO.ME3_3(), CO.ME3_4(), CO.ME3_5(), CO.ME3_6(), CO.ME3_7(), CO.ME3_8(), CO.ME3_9()}},
+      {{CO.ME4_1(), CO.ME4_2(), CO.ME4_3(), CO.ME4_4(), CO.ME4_5(), CO.ME4_6(), CO.ME4_7(), CO.ME4_8(), CO.ME4_9()}}
+      }};
+    for (int i = 0; i < 5; i++) {
+      for (int j = 0; j < 9; j++) {
+        if (counters.at(i).at(j) != 0) mpcLinkErrors->Fill(j + 1 + offset, endcap * (i + 0.5), counters.at(i).at(j));
+        else mpcLinkGood->Fill(j + 1 + offset, endcap * (i  + 0.5));
+      }
+    }
+    if (CO.ME1n_3() == 1) mpcLinkErrors->Fill(1 + offset, endcap * 5.5);
+    if (CO.ME1n_6() == 1) mpcLinkErrors->Fill(2 + offset, endcap * 5.5);
+    if (CO.ME1n_9() == 1) mpcLinkErrors->Fill(3 + offset, endcap * 5.5);
+    if (CO.ME2n_3() == 1) mpcLinkErrors->Fill(4 + offset, endcap * 5.5);
+    if (CO.ME2n_9() == 1) mpcLinkErrors->Fill(5 + offset, endcap * 5.5);
+    if (CO.ME3n_3() == 1) mpcLinkErrors->Fill(6 + offset, endcap * 5.5);
+    if (CO.ME3n_9() == 1) mpcLinkErrors->Fill(7 + offset, endcap * 5.5);
+    if (CO.ME4n_3() == 1) mpcLinkErrors->Fill(8 + offset, endcap * 5.5);
+    if (CO.ME4n_9() == 1) mpcLinkErrors->Fill(9 + offset, endcap * 5.5);
+    if (CO.ME1n_3() == 0) mpcLinkGood->Fill(1 + offset, endcap * 5.5);
+    if (CO.ME1n_6() == 0) mpcLinkGood->Fill(2 + offset, endcap * 5.5);
+    if (CO.ME1n_9() == 0) mpcLinkGood->Fill(3 + offset, endcap * 5.5);
+    if (CO.ME2n_3() == 0) mpcLinkGood->Fill(4 + offset, endcap * 5.5);
+    if (CO.ME2n_9() == 0) mpcLinkGood->Fill(5 + offset, endcap * 5.5);
+    if (CO.ME3n_3() == 0) mpcLinkGood->Fill(6 + offset, endcap * 5.5);
+    if (CO.ME3n_9() == 0) mpcLinkGood->Fill(7 + offset, endcap * 5.5);
+    if (CO.ME4n_3() == 0) mpcLinkGood->Fill(8 + offset, endcap * 5.5);
+    if (CO.ME4n_9() == 0) mpcLinkGood->Fill(9 + offset, endcap * 5.5);
   }
 
-  // Hits (LCTs)
-  edm::Handle<l1t::EMTFHit2016Collection> HitCollection;
+  // Hits (CSC LCTs and RPC hits)
+  edm::Handle<l1t::EMTFHitCollection> HitCollection;
   e.getByToken(hitToken, HitCollection);
-
-  for (std::vector<l1t::EMTFHit2016>::const_iterator Hit = HitCollection->begin(); Hit != HitCollection->end(); ++Hit) {
-    int endcap = Hit->Endcap();
-    int sector = Hit->Sector();
+  
+  // Maps CSC station and ring to the monitor element index and uses symmetry of the endcaps    
+  std::map<std::pair<int,int>,int> histIndexCSC = { {{1,4}, 9}, {{1,1}, 8}, {{1,2}, 7}, {{1,3}, 6},
+						    {{2,1}, 5}, {{2,2}, 4}, 
+						    {{3,1}, 3}, {{3,2}, 2},
+						    {{4,1}, 1}, {{4,2}, 0} };
+  
+  // Maps RPC staion and ring to the monitor element index and uses symmetry of the endcaps
+  std::map<std::pair<int, int>, int> histIndexRPC = { {{4,3}, 0}, {{4,2}, 1}, {{3,3}, 2}, {{3,2}, 3}, {{2,2}, 4}, {{1,2}, 5} };
+
+  for (std::vector<l1t::EMTFHit>::const_iterator Hit = HitCollection->begin(); Hit != HitCollection->end(); ++Hit) {
```

Comment:
Hi David,
This syntax produces the following error:
error: no matching function for call to 'begin(edm::Handle<std::vector<l1t::EMTFHit> >&)'
for (const auto & Hit: HitCollection)

I realize the current syntax isn't the most concise, but you have any other suggestions?
Best,
Andrew